standard_contracts/src/foreign_contracts.rs

```diff
@@ -77,12 +102,87 @@ pub mod core {
             ) -> Arguments<'a> {
                 result!()
             }
+
+            pub fn new_v1_formatted<'a>(
+                pieces: &'a [&'a str],
+                _args: &'a [ArgumentV1<'a>],
+                fmt: &'a [rt::v1::Argument],
+            ) -> Arguments<'a> {
+                precondition!(pieces.len() >= fmt.len());
```

Comment:
There is not much value in this contract either, since the Rust compiler already checks that there is an argument for every piece of the format that requires an argument.